graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -258,8 +260,11 @@ public void increment() {
         }
 
         public boolean isMatched() {
-            // If a stream has multiple stream rules, all of the rules have to match.
-            return ruleCount == matches;
+            switch (stream.getMatchingType()) {
```

Comment:
Looks like it's getting inlined. Sorry for the noise, just want to understand it. :smiley: 
```
                            @ 197   org.graylog2.streams.StreamRouterEngine$StreamMatch::isMatched (91 bytes)   inline (hot)
                              @ 7   org.graylog2.streams.StreamImpl::getMatchingType (28 bytes)   inline (hot)
                               \-> TypeProfile (2700/2700 counts) = org/graylog2/streams/StreamImpl
                                @ 6   java.util.LinkedHashMap::get (25 bytes)   inline (hot)
                                 \-> TypeProfile (2609/2609 counts) = java/util/LinkedHashMap
                                  @ 2   java.util.HashMap::getEntry (86 bytes)   inline (hot)
                                    @ 19   java.util.HashMap::hash (55 bytes)   inline (hot)
                                      @ 27   java.lang.String::hashCode (55 bytes)   inline (hot)
                                    @ 33   java.util.HashMap::indexFor (6 bytes)   inline (hot)
                                    @ 68   java.lang.String::equals (81 bytes)   (intrinsic)
                                  @ 17   java.util.LinkedHashMap$Entry::recordAccess (35 bytes)   inline (hot)
                                    @ 6   java.util.LinkedHashMap::access$000 (5 bytes)   accessor
                                    @ 23   java.util.LinkedHashMap$Entry::remove (23 bytes)   inline (hot)
                                    @ 28   java.util.LinkedHashMap::access$100 (5 bytes)   accessor
                                    @ 31   java.util.LinkedHashMap$Entry::addBefore (30 bytes)   inline (hot)
                                @ 24   org.graylog2.plugin.streams.Stream$MatchingType::valueOf (11 bytes)   inline (hot)
                                  @ 4   java.lang.Enum::valueOf (73 bytes)   inline (hot)
                                    @ 1   java.lang.Class::enumConstantDirectory (114 bytes)   too big
                              @ 12   java.lang.Enum::ordinal (5 bytes)   accessor
```